### ISSUE TITLE
[TS] Property 'shift' of type 'string[] | undefined' is not assignable to string index type 'string[]

### ISSUE THREAD

**Issue by neXorianus** [hodgef/simple-keyboard#68]
**Simple-keyboard version**
"version": "2.21.1",

**Describe the bug**
Getting type-error in a typescript environment:
```
ERROR in .../node_modules/simple-keyboard/build/index.d.ts
4:5 Property 'shift' of type 'string[] | undefined' is not assignable to string index type 'string[]'.
    2 |   interface KeyboardLayoutObject {
    3 |     default: string[];
  > 4 |     shift?: string[];
      |     ^
    5 |     [key: string]: string[];
    6 |   }
    7 | 
```
making shift not optional removes the error - but I don't know if this is the best or even correct way to go.

**Comment by hodgef** [comment]
Hello @neXorianus ,

Could you elaborate on what is triggering that error? I can see the Angular typescript demo working well here: https://codesandbox.io/s/y5pkc

A code sample, repo or sandbox would help! I'll reopen once that's provided.

Regards,
Francisco Hodge

**Comment by neXorianus** [comment]
I'm using the repo with a vue-cli3 with typescript installation.
A simple `yarn serve|build` brings up the error.
I try to create a test repo for you, but that might need to wait until next week.

**Comment by hodgef** [comment]
Thanks for the details, those steps to reproduce should enough. I'll take a look and update the ticket soon.

**Comment by neXorianus** [comment]
reproduce:

`vue create temp_app`
choose 
```
? Please pick a preset: Manually select features
? Check the features needed for your project: Babel, TS, Router, Vuex, CSS Pre-processors, Linter
? Use class-style component syntax? Yes
? Use Babel alongside TypeScript for auto-detected polyfills? Yes
? Use history mode for router? (Requires proper server setup for index fallback in production) No
? Pick a CSS pre-processor (PostCSS, Autoprefixer and CSS Modules are supported by default): Sass/SCSS (with dart-sass)
? Pick a linter / formatter config: TSLint
? Pick additional lint features: (Press <space> to select, <a> to toggle all, <i
> to invert selection)Lint on save
? Where do you prefer placing config for Babel, PostCSS, ESLint, etc.? In dedicated config files
```
then `cd tmp_app && yarn add simple-keyboard`
then add `import Keyboard from 'simple-keyboard';` to `src/views/Home.vue`
results in Type error

**Comment by hodgef** [comment]
Hello @neXorianus,

This should be fixed as of 2.21.2, please feel free to update and try it out.

Sadly though, it involves setting the type to `any` temporarily, as I didn't find a better way to get rid of the error. If you come across a better solution please feel free to share it and I will update the types.

Commit: https://github.com/hodgef/simple-keyboard/commit/a0c24db9d24c49ee57aae56a8ce2055823b005b3

Regards,
Francisco Hodge
